repose-aggregator/components/filters/herp-filter/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala

```diff
@@ -122,7 +122,11 @@ class HerpFilter @Inject()(configurationService: ConfigurationService,
     }
 
     val projectIds = httpServletRequest.getHeaders(OpenStackServiceHeader.TENANT_ID.toString).asScala
-      .++(httpServletRequest.getHeaders(X_PROJECT_ID).asScala).toTraversable
+      .++(httpServletRequest.getHeaders(X_PROJECT_ID).asScala).toTraversable match {
+      case x if x.nonEmpty => x
```

Comment:
I did update these a bit to abstract out to the common `HeaderInteractor` interface on the new `Wrapper`'s. That way some of the copy/paste could be eliminated. Lines wise I think it was the same, but it looks nicer now.